exchanges/binanceus/binanceus_wrapper.go

```diff
@@ -0,0 +1,1011 @@
+	pairs, err := bi.FetchTradablePairs(ctx, asset.Spot)
+	if err != nil {
+		return err
+	}
+	p, err := currency.NewPairsFromStrings(pairs)
+	if err != nil {
+		return err
+	}
+	return bi.UpdatePairs(p, asset.Spot, false, forceUpdate)
+}
+
+// UpdateTicker updates and returns the ticker for a currency pair
+func (bi *Binanceus) UpdateTicker(ctx context.Context, p currency.Pair, a asset.Item) (*ticker.Price, error) {
+	switch a {
+	case asset.Spot:
+		tick, err := bi.GetPriceChangeStats(ctx, p)
+		if err != nil {
+			return nil, err
+		}
+		err = ticker.ProcessTicker(&ticker.Price{
+			Last:         tick.LastPrice,
+			High:         tick.HighPrice,
+			Low:          tick.LowPrice,
+			Bid:          tick.BidPrice,
+			Ask:          tick.AskPrice,
+			Volume:       tick.Volume,
+			QuoteVolume:  tick.QuoteVolume,
+			Open:         tick.OpenPrice,
+			Close:        tick.PrevClosePrice,
+			Pair:         p,
+			ExchangeName: bi.Name,
+			AssetType:    a,
+		})
+		if err != nil {
+			return nil, err
+		}
+	default:
+		return nil, fmt.Errorf("assetType not supported: %v", a)
+	}
+	return ticker.GetTicker(bi.Name, p, a)
+}
+
+// UpdateTickers updates all currency pairs of a given asset type
+func (bi *Binanceus) UpdateTickers(ctx context.Context, a asset.Item) error {
+	switch a {
+	case asset.Spot:
+		tick, err := bi.GetTickers(ctx)
+		if err != nil {
+			return err
+		}
+
+		pairs, err := bi.GetEnabledPairs(a)
+		if err != nil {
+			return err
+		}
+		for i := range pairs {
+			for y := range tick {
+				pairFmt, err := bi.FormatExchangeCurrency(pairs[i], a)
+				if err != nil {
+					return err
+				}
+				if tick[y].Symbol != pairFmt.String() {
+					continue
+				}
+				err = ticker.ProcessTicker(&ticker.Price{
+					Last:         tick[y].LastPrice,
+					High:         tick[y].HighPrice,
+					Low:          tick[y].LowPrice,
+					Bid:          tick[y].BidPrice,
+					Ask:          tick[y].AskPrice,
+					Volume:       tick[y].Volume,
+					QuoteVolume:  tick[y].QuoteVolume,
+					Open:         tick[y].OpenPrice,
+					Close:        tick[y].PrevClosePrice,
+					Pair:         pairFmt,
+					ExchangeName: bi.Name,
+					AssetType:    a,
+				})
+				if err != nil {
+					return err
+				}
+			}
+		}
+	default:
+		return fmt.Errorf("assetType not supported: %v", a)
+	}
+	return nil
+}
+
+// FetchTicker returns the ticker for a currency pair
+func (bi *Binanceus) FetchTicker(ctx context.Context, p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
+	fpairs, er := bi.FormatExchangeCurrency(p, assetType)
+	if er != nil {
+		return nil, er
+	}
+	bi.appendOptionalDelimiter(&fpairs)
+
+	tickerNew, err := ticker.GetTicker(bi.Name, p, assetType)
+	if err != nil {
+		return bi.UpdateTicker(ctx, p, assetType)
+	}
+	return tickerNew, nil
+}
+
+// appendOptionalDelimiter ensures that a delimiter is
+// present for long character currencies
+func (b *Binanceus) appendOptionalDelimiter(p *currency.Pair) {
+	if len(p.Quote.String()) > 3 ||
```

Comment:
NOTE FOR ME: Len method could be added.

Reply:
I was thinking of adding a `length()` or `Len()`method to `currency.Code` type, when I get some time to start going over this trello card: [here](https://trello.com/c/54eizOIo/143-currency-package-upgrades). But for now other priorities. It's out of scope for your PR.